test/plugin/test_out_buffered_stdout.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../helper'
+require 'fluent/test'
+require 'fluent/plugin/out_buffered_stdout'
+
+class BufferedStdoutOutputTest < Test::Unit::TestCase
+  def setup
+    Fluent::Test.setup
+  end
+
+  CONFIG = %[
+  ]
+
+  def create_driver(conf = CONFIG)
+    Fluent::Test::BufferedOutputTestDriver.new(Fluent::BufferedStdoutOutput).configure(conf)
+  end
+
+  def test_configure
+    d = create_driver
+    assert_equal 'json', d.instance.output_type
+  end
+
+  def test_configure_output_type
+    d = create_driver(CONFIG + "\noutput_type json")
+    assert_equal 'json', d.instance.output_type
+
+    d = create_driver(CONFIG + "\noutput_type hash")
+    assert_equal 'hash', d.instance.output_type
+
+    assert_raise(Fluent::ConfigError) do
+      d = create_driver(CONFIG + "\noutput_type foo")
+    end
+  end
+end
```

Comment:
In tests, logger is configured with `Fluent::Test::DummyLogDevice`. It stores logs in an array, and these logs can be accessed by `driver.instance.log.out.logs`.
https://github.com/fluent/fluentd/blob/master/lib/fluent/test/base.rb#L95

I'm doing it in a pull-request diff.
https://github.com/fluent/fluentd/pull/843/files#diff-a741a1b1e75fe157fb8890bdd1a7040dR180